pages/Graph/Overview.md

```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
+be able to determine who the follower(DSNP Id doing the following) and followee(DSNP Id being followed) are for a given event.
+
+## Graph Change Announcement Format
+Anytime a user "follows" or "unfollows" someone - a graph change event is created.
+A graph change will contain the following information: `fromId`, `changeType`, `objectId`, `nonce`. See [Messages Overview](/Messages/Overview) for more information on these fields.
+
+## Graph Storage
+All graph change events are signed and added to a [batch file](/Batches/Overview).
+Once that batch file is complete, it is [announced](/Messages/Announce) on the blockchain per the DSNP specification. 
+Each announce event specifies the type of ([dsnpType](/Messages/Announce)) messages that are in the batch file. 
+
+## Graph Retrieval, Ordering & Reading
+To have an accurate representation of the social graph for a given DSNP Identity - it is necessary to retrieve the entire graph from the chain.
+
+To retrieve the graph - do the following:
+1. Retrieve all the log events[dsnpType](/Messages/Types)`GraphChange` from the chain
+1. Retrieve the batch file from each log event. Each log event of type [GraphChange](/Messages/Announce) has a field called `dsnpUri` which contains a uri pointing to a [batch file](/Batches/Overview). 
+1. Query the batch files for the data for a particular DSNP Id to retrieve information about the respective graph. For more on how batch file storage and how to query the batch file see - [batches overview](/Batches/Overview)
+1. Order the retrieved data based on the following  
+    1. Block Number Ascending
+    1. Transaction Index Ascending
+    1. Log Index Ascending
+    1. Batch File Row Index Ascending
+                                                
+For more on ordering see [Message Ordering Specification](/Messages/Ordering).
+
+#### Reading the graph
+Once the retrieval of the entire graph is complete and data is ordered, read the ordered data to see the most recent state of the graph between two respective DSNP Ids (i.e the last graph change event shows Bob has unfollowed Charlie)
+
+If the entire graph has been retrieved and stored, it is possible to then retrieve new events from the point where the last graph ended. This can be done by retrieving `GraphChange` log events from the chain starting with a specific block number. 
+For example: The graph for a given user has been retrieved and stored up until block number 18. A filter can be added to the logs being retrieved from the chain to only retrieve log events starting from block 19.
+
+## Replay Attacks
+
+[Replay attacks](https://en.wikipedia.org/wiki/Replay_attack) are prevented by ensuring that each graph change event is identifiably unique. 
+To allow for uniqueness, each graph change event has a nonce. This nonce field is timestamp represented as microseconds since Unix epoch.
```

Comment:
```suggestion
To allow for uniqueness, each graph change event has a nonce.
This nonce field is timestamp represented as microseconds since Unix epoch.
```